auth-api/src/main/java/io/stargate/auth/api/resources/AuthResource.java

```diff
@@ -232,16 +252,53 @@ public Response createTokenFromUsername(
     } catch (UnauthorizedException e) {
       return Response.status(Response.Status.UNAUTHORIZED)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoStore)
           .build();
     } catch (Exception e) {
       logger.error("Failed to create token", e);
       return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoStore)
           .build();
     }
 
     return Response.status(Response.Status.CREATED)
         .entity(new AuthTokenResponse().authToken(token))
+        .cacheControl(cacheControlWithMaxAge)
         .build();
   }
+
+  /**
+   * Used to create a Cache-Control header containing the proper values for the token's max-age and
+   * s-maxage as defined by {@code stargate.auth_tokenttl} minus some time to avoid expiration. If
+   * the property is not set then the default value of 1790 will be used.
+   *
+   * @return A {@link CacheControl} object which is converted into a Cache-Control header with
+   *     proper directives.
+   */
+  private CacheControl cacheControlWithAge() {
+    // For a large enough TTL it's safe to knock off 10 seconds but for a smaller number reducing by
+    // that much will negatively impact caching so instead subtract only 1 second.
```

Comment:
maybe instead of this logic, we should validate that `tokenMaxAge` is set to some sane value? And use `- 10` always?

Reply:
If you're using a TTL of 5 seconds for creating tokens then we have to respect that here as well but we still need to allow some padding for users to be able to generate an new token before the old expires